android/src/org/ppsspp/ppsspp/NativeActivity.java

```diff
@@ -309,71 +286,66 @@ public void checkForVibrator() {
+
+	public native void runEGLRenderLoop(Surface surface);
+	// Tells the render loop thread to exit, so we can restart it.
+	public native void exitEGLRenderLoop();
+
+	void updateDisplayMetrics(Point outSize) {
+		DisplayMetrics metrics = new DisplayMetrics();
+		Display display = getWindowManager().getDefaultDisplay();
+		display.getMetrics(metrics);
+
+		float refreshRate = display.getRefreshRate();
+		if (outSize == null) {
+			outSize = new Point();
+		}
+		GetScreenSize(outSize);
+		NativeApp.setDisplayParameters(outSize.x, outSize.y, metrics.densityDpi, refreshRate);
 	}
 
-    @Override
+	@Override
     public void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState); 
+		super.onCreate(savedInstanceState);
 		registerCallbacks();
     	installID = Installation.id(this);
 
+    	updateDisplayMetrics(null);
+
 		if (!initialized) {
 			Initialize();
 			initialized = true;
 		}
+
+		// OK, config should be initialized, we can query for screen rotation.
+		if (Build.VERSION.SDK_INT >= 9) {
+			updateScreenRotation();
+		}
+
 		// Keep the screen bright - very annoying if it goes dark when tilting away
 		Window window = this.getWindow();
 		window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
 		setVolumeControlStream(AudioManager.STREAM_MUSIC);
-		
+
 		gainAudioFocus(this.audioManager, this.audioFocusChangeListener);
         NativeApp.audioInit();
-        
-        mGLSurfaceView = new NativeGLView(this);
-		nativeRenderer = new NativeRenderer(this);
-
-		Point sz = new Point();
-		getDesiredBackbufferSize(sz);
-		if (sz.x > 0) {
-			Log.i(TAG, "Requesting fixed size buffer: " + sz.x + "x" + sz.y);
-			// Auto-calculates new DPI and forwards to the correct call on mGLSurfaceView.getHolder()
-			nativeRenderer.setFixedSize(sz.x, sz.y, mGLSurfaceView);
-		}
-        mGLSurfaceView.setEGLContextClientVersion(2);
-        
-        // Setup the GLSurface and ask android for the correct 
-        // Number of bits for r, g, b, a, depth and stencil components
-        // The PSP only has 16-bit Z so that should be enough.
-        // Might want to change this for other apps (24-bit might be useful).
-        // Actually, we might be able to do without both stencil and depth in
-        // the back buffer, but that would kill non-buffered rendering.
-        
-        // It appears some gingerbread devices blow up if you use a config chooser at all ????  (Xperia Play)
-        //if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
-        
-        // On some (especially older devices), things blow up later (EGL_BAD_MATCH) if we don't set the format here,
-        // if we specify that we want destination alpha in the config chooser, which we do.
-        // http://grokbase.com/t/gg/android-developers/11bj40jm4w/fall-back
-        
-        
-        // Needed to avoid banding on Ouya?
-        if (Build.MANUFACTURER == "OUYA") {
-        	mGLSurfaceView.getHolder().setFormat(PixelFormat.RGBX_8888);
```

Comment:
I hope so, I realized I forgot to dig out the Ouya so I'll check and see if I still get banding.